### ISSUE TITLE
Scraper does not get the "Default (Aired Order)" order

### ISSUE THREAD

**Issue by Hlsgs** [thetvdb/metadata.tvshows.thetvdb.com.v4.python#23]
Episodes and Specials are correctly scraped, but all specials are only placed as such, and not in between the episodes they are supposed to be between. I am only interested and refering to TVDBs "Default (Aired Order)", which is the default setting for the scraper.

This seems to happen for any and all series, but, if need be, I can provide a dummy file structure to replicate on and the resulting log.

**Comment by KarellenX** [comment]
This issue was discussed on the forum here... https://forum.kodi.tv/showthread.php?tid=370232

It relates to the Specials use of the "Airs Before", "Airs After" data. The scraper seems to ignore these fields, so the Specials are not sorted into their correct order in the normal Seasons.

I also confirmed this issue.

**Comment by antheaezzell** [comment]
@Hlsgs and @KarellenX this has been ticketed - https://mediamorph.atlassian.net/browse/TVD-3479 feel free to update the forums to let them know we are aware of the bug and are working to prioritize the work.

**Comment by Hlsgs** [comment]
@antheaezzell Quick question: is this purely on the API side or will there need to be an update to the KODI scraper as well?

**Comment by antheaezzell** [comment]
No, I think the API is treating this properly or we would have gotten a lot of reports of it in the v4 repo. It's possible that the scraper needs an update @Hlsgs

**Comment by Hlsgs** [comment]
No change in v1.1.2

**Comment by antheaezzell** [comment]
This is still in progress and was not part of the most recent release @Hlsgs

**Comment by maxifwortman** [comment]
@Hlsgs Yes, 1.1.3 fix that issue.

**Comment by Hlsgs** [comment]
@maxifwortman Tested 1.1.3 and specials placement is all over the place. Taking [BSG](https://thetvdb.com/series/battlestar-galactica-2003) as an example:

- [the pilot miniseries](https://thetvdb.com/series/battlestar-galactica-2003/episodes/117847), listed as "Airs before season 1, episode 1", gets added at the beginning of each and every season
- both [The Resistance](https://thetvdb.com/series/battlestar-galactica-2003/episodes/308518), listed as "Airs after season 2" and [Razor](https://thetvdb.com/series/battlestar-galactica-2003/episodes/331114), listed as "Airs before season 4, episode 1" get added before S03E01
- [Face of the Enemy](https://thetvdb.com/series/battlestar-galactica-2003/episodes/416926), listed as "Airs before season 4, episode 11" gets added after S03E09
- [The Plan](https://thetvdb.com/series/battlestar-galactica-2003/episodes/1197791) and [Blood & Chrome](https://thetvdb.com/series/battlestar-galactica-2003/episodes/7782880), both listed as  "Airs after season 4", do not get added at all withing the seasons.

**Comment by KarellenX** [comment]
I can confirm the issue by @Hlsgs using Battlestar Galactica.

In the database: https://kodi.wiki/view/Databases/MyVideos#episode
**c15**=Season Number- Specials Sorting
**c16**=Episode Number- Specials Sorting
With some episodes the database has been filled with **0** values instead of the correct Sort Season and Sort Episode number. See image.

Looking at the log, the API seems to return two types of values that I think is connected with Sorting (I am not overly familiar with your API)...
**'sortseason': 0, 'sortepisode': 0,** 
and
**'airsBeforeSeason': 1, 'airsBeforeEpisode': 1**

It seems that the **sortx** is being used for the data and some of those entries have invalid "0" value **sortx** 

Is the API returning invalid data?

Log... https://paste.kodi.tv/sawitukuca.kodi
Line 1776 for sortseason, sortepisode
Line 1777 for airsbefore data

![sortseason](https://user-images.githubusercontent.com/44626605/206878031-b88c8690-2ef7-45d0-8338-0f32a9a5b872.jpg)

**Comment by antheaezzell** [comment]
Hi @Hlsgs and @KarellenX we are looking at these issues right now. I will get back to you with an update.

**Comment by antheaezzell** [comment]
@Hlsgs I believe this work has been done in our most recent release. Please let me know if you are still encountering issues.

**Comment by Hlsgs** [comment]
There's one more issue I can find: specials that are listed as "Airs after season X" are listed at the beginning of season X+1 instead. As a consequence(and this is the main problem), specials that follow the end of the show are not listed at the end of the last season. That used to be the case with the old scrapers and it helped with not having to seek additional info online. Still taking [BSG](https://thetvdb.com/series/battlestar-galactica-2003) as an example, this applies to the following:

- listed at the beginning of the next season:  [The Resistance](https://thetvdb.com/series/battlestar-galactica-2003/episodes/308518)
- not listed in the flow of the show, because they are after the end of the show: [The Plan](https://thetvdb.com/series/battlestar-galactica-2003/episodes/1197791) and [Blood & Chrome](https://thetvdb.com/series/battlestar-galactica-2003/episodes/7782880)

Interestingly, thetvdb has a separate type of listing as "Airs before season X, episode 1", which now produces the same result as "Airs after season X-1". Example: [Razor](https://thetvdb.com/series/battlestar-galactica-2003/episodes/331114), listed as "Airs before season 4, episode 1"

@KarellenX can you confirm?

**Comment by antheaezzell** [comment]
This part still appears to be a problem. I'm going to reticket. Internal ticket here - https://mediamorph.atlassian.net/browse/TVD-3571

**Comment by Hlsgs** [comment]
@antheaezzell Any ETA of this?

**Comment by antheaezzell** [comment]
Hi @Hlsgs it's probably going to be several weeks before we can get to this. It is our highest priority, but we only have one engineer who knows the plugin and he's got a few other work commitments on a different project. It's my top priority when I get him back.

**Comment by antheaezzell** [comment]
@Hlsgs can you test? This work was released - https://github.com/thetvdb/metadata.tvshows.thetvdb.com.v4.python/releases/tag/1.1.5
